src/main/drivers/video_max7456.c

```diff
@@ -0,0 +1,620 @@
+    0x34, 0x35, 0x36, 0x37, 0x38, 0x39, 0x3A, 0x3B, 0x3C, 0x3D, 0x3E, 0x00, 0x00, 0x00, 0x00, 0x00, // 112 - 127  "pqrstuvwxyz{|}~ "
+};
+
+static const uint8_t max7456_defaultFont_fontToASCIIMapping[] = {
+    ' ', '1', '2', '3', '4', '5', '6', '7', '8', '9', '0', 'A', 'B', 'C', 'D', 'E', // 0x00 - 0x0F
+    'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', // 0x10 - 0x1F
+    'V', 'W', 'X', 'Y', 'Z', 'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', // 0x20 - 0x2F
+    'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', '(', // 0x30 - 0x3F
+    ')', '.', '?', ';', ':', ',', '\'', '/', '"', '-', '<','>', '@'                 // 0x40 - 0x4F
+};
+#endif
+
+
+static bool max7456ExtiHandler(const extiConfig_t *extiConfig)
+{
+    if (EXTI_GetITStatus(extiConfig->exti_line) == RESET) {
+        return false;
+    }
+
+    EXTI_ClearITPendingBit(extiConfig->exti_line);
+
+    return true;
+}
+
+
+void max7456_updateLOSState(void)
+{
+    // "LOS goes high when the VIN sync pulse is lost for 32 consecutive lines. LOS goes low when 32 consecutive valid sync pulses are received."
+    uint8_t status = GPIO_ReadInputDataBit(max7456LOSExtiConfig->gpioPort, max7456LOSExtiConfig->gpioPin);
+
+    max7456State.los = status != 0;
+
+    //debug[0] = max7456State.los;
+}
+
+void LOS_EXTI_Handler(void)
+{
+    static uint32_t callCount = 0;
+    bool set = max7456ExtiHandler(max7456LOSExtiConfig);
+    callCount++;
+    if (!set) {
+        return;
+    }
+
+    max7456State.losCounter++;
+
+    max7456_updateLOSState();
+}
+
+void VSYNC_EXTI_Handler(void)
+{
+    static uint32_t callCount = 0;
+    bool set = max7456ExtiHandler(max7456VSYNCExtiConfig);
+    callCount++;
+    if (!set) {
+        return;
+    }
+
+    max7456State.vSyncDetected = true;
+
+    max7456State.frameCounter++;
+    if (max7456State.lineCounter > max7456State.maxLinesDetected) {
+        max7456State.maxLinesDetected = max7456State.lineCounter;
+    }
+    max7456State.lineCounter = 0;
+
+    /* Time between VSYNC pulses can be measured using the code below.
+     * NTSC: ~16673us (60 FPS), PAL: ~19990us (50FPS)
+     */
+    /*
+    static bool shouldStartMeasuring = true;
+
+    if (shouldStartMeasuring) {
+        TIME_SECTION_BEGIN(2);
+    } else {
+        TIME_SECTION_END(2);
+    }
+    shouldStartMeasuring = !shouldStartMeasuring;
+    */
+
+
+
+    //debug[1] = max7456State.frameCounter;
+    //debug[3] = max7456State.maxLinesDetected;
+}
+
+void HSYNC_EXTI_Handler(void)
+{
+    static uint32_t callCount = 0;
+    bool set = max7456ExtiHandler(max7456HSYNCExtiConfig);
+    callCount++;
+    if (!set) {
+        return;
+    }
+
+    max7456State.hSyncDetected = true;
+
+
+    max7456State.lineCounter++;
+
+    //debug[2] = max7456State.lineCounter;
+}
+
+
+typedef void (*handlerFuncPtr)(void);
```

Comment:
I have my f3 disco now,  is there schematics for the MAX chip circuit? Can i mod a minim osd?  Or do i design and reverse engineer from scratch?





-------- Original message --------
From: Dominic Clifton <notifications@github.com>
Date: 2016-07-02  12:32 PM  (GMT-06:00)
To: cleanflight/cleanflight <cleanflight@noreply.github.com>
Subject: Re: [cleanflight/cleanflight] First cut of OSD support (#2226)

> +    bool set = max7456ExtiHandler(max7456HSYNCExtiConfig);
> +    callCount++;
> +    if (!set) {
> +        return;
> +    }
> +
> +    max7456State.hSyncDetected = true;
> +
> +
> +    max7456State.lineCounter++;
> +
> +    //debug[2] = max7456State.lineCounter;
> +}
> +
> +
> +typedef void (*handlerFuncPtr)(void);

I opened an issue #2191 to discuss stuff like this.  If it's not already added then I will add it to that issue.

---
You are receiving this because you are subscribed to this thread.
Reply to this email directly or view it on GitHub:
https://github.com/cleanflight/cleanflight/pull/2226/files/997837c190ffbec1514c50bab188c47ba9cd3b69#r69381324